src/Infrastructure.VS/Roslyn/IConnectedModeRoslynAnalyzerProvider.cs

```diff
@@ -0,0 +1,42 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using SonarLint.VisualStudio.Core.Binding;
+
+namespace SonarLint.VisualStudio.Infrastructure.VS.Roslyn;
+
+/// <summary>
+/// Returns SonarAnalyzer.CSharp & SonarAnalyzer.VisualBasic analyzer DLLs that are downloaded from the server for the current binding
+/// </summary>
+public interface IConnectedModeRoslynAnalyzerProvider
```

Comment:
Consider if Repository might be a better name

Reply:
I would like to keep the provider and the writer interfaces split. The implementation could have Repository in the name though